cabal-install/Distribution/Client/GlobalFlags.hs

```diff
@@ -219,27 +216,32 @@ initSecureRepo :: Verbosity
                -> (SecureRepo -> IO a)  -- ^ Callback
                -> IO a
 initSecureRepo verbosity httpLib RemoteRepo{..} cachePath = \callback -> do
-    withRepo $ \r -> do
-      requiresBootstrap <- Sec.requiresBootstrap r
+    requiresBootstrap <- withRepo [] Sec.requiresBootstrap
+
+    mirrors <- if requiresBootstrap
```

Comment:
It seems worth mentioning (in info) that we're querying DNS to get some mirrors. And it'll only happen once.